server/src/main/java/org/elasticsearch/common/unit/RelativeByteSizeValue.java

```diff
@@ -47,11 +47,16 @@ public RatioValue getRatio() {
      * Calculate the size to use, optionally catering for a max headroom.
      * @param total the total size to use
      * @param maxHeadroom the max headroom to cater for or null (or -1) to ignore.
+     * @param floorInsteadOfCeil if true, and in case a ratio/percentage value exists, then when calculating ratio * total bytes, use
+     *                           Math.floor() instead of Math.ceil(). This is used for example in case of trying to calculate the minimum
+     *                           free bytes for staying lower than the ratio, where one would need to subtract the result of this function
+     *                           from the total bytes.
      * @return the size to use
      */
-    public ByteSizeValue calculateValue(ByteSizeValue total, ByteSizeValue maxHeadroom) {
+    public ByteSizeValue calculateValue(ByteSizeValue total, ByteSizeValue maxHeadroom, boolean floorInsteadOfCeil) {
```

Comment:
I find this boolean parameter makes it pretty hard to read the calling code. Does it make more sense to capture the rounding behaviour as an intrinsic property of the `RelativeByteSizeValue` itself? If not, could we have two methods with more descriptive names?

Reply:
Hmm, as long as calculateValue() is inside RelativeByteSizeValue class, I found it OK to handle the rounding behavior here, since I wanted the floor instead of ceil in one specific case (which is in the comments of the function). Using two methods is also nice. I named them `calculateValueWithCeil` and `calculateValueWithFloor`. Pushed a commit. Feel free to tell me if you have other suggestions and unresolve this conversation.